Lib/lib2to3/refactor.py

```diff
@@ -466,22 +468,31 @@ def refactor_tree(self, tree, name):
             fixer.finish_tree(tree, name)
         return tree.was_changed
 
-    def traverse_by(self, fixers, traversal):
+    def traverse_by(self, fixers, tree):
```

Comment:
I expect your overall logic is sound in this patch.  But this is a public API change, and while we've typically exempted lib2to3 from needing to concern itself about deprecations and API changes by leaving it undocumented and explicitly promising not to care about this module in that sense... Enough things use it today that we should probably not do that anymore.

Especially given that we're considering it a deprecated library these days.  It isn't likely to stand up well to 3.10+'s new grammar.

Various projects have forked lib2to3 into their own maintained thing either directly on PyPI or as an internalized third_party fork within their own project (as I believe Black has done?).  It is probably best to do that for changes of this nature rather than push for them to be included in a future stdlib lib2to3.

https://bugs.python.org/issue40360  for context.